### PR TITLE
fix(elasticSearchDomain): add cloudwatchLogs, cognitoIdentityPool, cognitoUserPool, iamRole connections

### PR DESCRIPTION
**Pull request by hjaraujof**
## Issue tracker links

Handle TODOs for elasticSearchDomain
[CG-1066](https://autoclouddev.atlassian.net/browse/CG-1066?atlOrigin=eyJpIjoiOTcwYzkzYTg3MTc2NDFiZWI3ZDljYzg2OGI4MmUwOWIiLCJwIjoiaiJ9)

## Changes/solution

Adds new connections to `elasticSearchDomain` and the following services

- `cloudwatchLogs`
- `cognitoIdentityPool`
- `cognitoUserPool`
- `iamRole`

## Testing

Needs an OpenSearch domain in the sandbox account, please dm me when you are ready to test. I can spin a new one, the one used for the developing process has been torn down.

### CODE REVIEW

File: src/services/elasticSearchDomain/connections.ts [hunk 1]
```diff
@@ -20,9 +26,14 @@ export default ({
     DomainId,
     VPCOptions: { SecurityGroupIds = [] } = {},
     EncryptionAtRestOptions: { KmsKeyId } = {},
+    CognitoOptions: { IdentityPoolId, UserPoolId, RoleArn } = {},
+    LogPublishingOptions = {},
   } = domain
   const connections: ServiceConnection[] = []
-
+  const logGroupsArns: string[] = Object.entries(LogPublishingOptions).map(
+    ([, value]) => value.CloudWatchLogsLogGroupArn
```
Comment (tyler-dunkel) [inline src/services/elasticSearchDomain/connections.ts:34]: could value be undefined here?

File: src/services/elasticSearchDomain/connections.ts [hunk 2]
```diff
@@ -59,6 +70,91 @@ export default ({
...
+    ({ name }) => name === services.cloudwatchLog
+  )
+  if (cloudwatchLogGroups?.data?.[region]) {
+    const dataAtRegion: RawAwsLogGroup[] = cloudwatchLogGroups.data[
+      region
+    ].filter(
+      ({ arn }: RawAwsLogGroup) =>
+        !isEmpty(logGroupsArns) &&
+        logGroupsArns.filter(
+          str => `${str}:*`.includes(arn) // A small interpolation hack to be able to match the full arn
```
Comment (tyler-dunkel) [inline src/services/elasticSearchDomain/connections.ts:145]: hmm can you provide an example here of what str might be and what the arn might be?

File: src/services/elasticSearchDomain/connections.ts (hunk 1 above)
Comment (hjaraujof) [reply src/services/elasticSearchDomain/connections.ts:34]: No, an undefined value is something that cannot be reproduced and wouldn't make much sense to store. The object normally contains a map in which keys are always uppercase with a meaningful string that describes the purpose and the value contains the Cloudwatch log group name and ARN.

File: src/services/elasticSearchDomain/connections.ts (hunk 2 above)
Comment (hjaraujof) [reply src/services/elasticSearchDomain/connections.ts:145]: I found out that Cloudwatch log groups ARNs have that wildcard attached at the end.

Like this for example

Screenshot from 2022-04-14 14-33-25 (1)

**Review by tyler-dunkel — APPROVED** [review]

**Pull request merged**